src/index.ts

```diff
@@ -0,0 +1,41 @@
+import {Boom} from '@hapi/boom';
+import {Request, Response} from 'express';
+
+export class BoomableError extends Error {
+  public statusCode = 500;
+}
+
+type ExpressErrorMiddleware = (
+  err: Boom | BoomableError | Error,
+  req: Request,
+  res: Response,
+  next: (err: Error) => void,
+) => void;
+
+export default function (): ExpressErrorMiddleware {
+  return (
+    err: Boom | BoomableError | Error,
+    _req: Request,
+    res: Response,
+    _next: (err: Error) => void,
+  ) => {
+    if (!(err instanceof Error)) {
+      err = new Error(err);
+    }
+    let boom: Boom;
+    if (err instanceof Boom) {
+      boom = err;
+    } else {
+      let statusCode = 500;
+      if (err instanceof BoomableError) {
```

Comment:
Since no code ever instantiates `BoomableErrors`, this condition can never be true. Keep in mind that TypeScript uses duck typing (i.e. structural typing), but TypeScript is only used for typechecking. JavaScript `instanceof` uses prototypical inheritance, which is a completely separate concept. So this `instanceof` check will only be true for actual instances of this class (well, it's more nuanced than that, but that's the general idea).

I think this might all be moot anyway, because I don't think we need this code block. Here's the original CoffeeScript:
```coffee
    status = err.statusCode or err.status
    Boom.wrap(err, status)
```
So all we want is to pass some data into the boom call that is `err.statusCode` if defined, otherwise `err.status` if defined. (It probably isn't even necessary to support both, but let's ignore that to keep this PR focused on porting to TS).
I think we can achieve this with a simple statement like:
```js
    const statusCode = err.statusCode ?? err.status;
    boom = new Boom(err, {statusCode});
```
We don't need the 500 default because that's already boom's default.